src/api/chainApi/unique/marketController.ts

```diff
@@ -123,21 +125,23 @@ class MarketController implements IMarketController {
     }
   }
 
-  public async addToWhiteList(account: string, options: TransactionOptions): Promise<void> {
+  public async addToWhiteList(account: string, options: TransactionOptions, signMessage: { (message: string, account?: string | Account | undefined): Promise<string>; (arg0: string): any; }): Promise<void> {
     const ethAddress = getEthAccount(account);
     const isWhiteListed = await this.checkWhiteListed(ethAddress);
     if (isWhiteListed) {
       return;
     }
-    const minDeposit = this.kusamaApi?.consts.balances?.existentialDeposit;
-
-    const tx = this.kusamaApi.tx.balances.transfer(this.escrowAddress, minDeposit);
-    const signedTx = await options.sign(tx);
-
-    if (!signedTx) throw new Error('Breaking transaction');
+    const signaturePhrase = 'allowedlist';
+
+    let signature;
+      try {
+        signature = await signMessage(signaturePhrase);
+        await addToWhitelist({ account: account }, signature);
```

Comment:
:warning: 
marketController sohuld stay agnostic about outter world. No API calls inside of it.

You can pass customSend inside options instead (see "transferToken" method)